### ISSUE TITLE
[blazor-server] In a Blazor App with Authentication, the static files from ~/Identity/... are not resolvable in Production environment (404), which crashes layout and validation (css, js, ...)

### ISSUE THREAD

**Issue by tiwj** [dotnet/aspnetcore#20357]
### Description
When creating a new Blazor App with Authentication, the browser fails to resolve the ressources below the path ~/Identity/ when running in Production mode (works fine in Development mode).

It seems that something is wrong with the static file mappings for /Identity/ outside Development environment.

### To Reproduce
1. Create new BlazorApp with Authentication > Individual User Accounts > Store in-app
2. Start the application with Console or IISExpress
3. Go to the "Register" or "Login" page of the Identity framework by clicking the top right links
4. See page is rendering fine with correct layout (bootstrap etc.)
5. See network traffic of your browser resolving js/css etc from ~/Identity/ properly:
![grafik](https://user-images.githubusercontent.com/39725141/78028058-ad4f5900-735e-11ea-9911-cc087bf59711.png)
6. Stop application and go Properties > LaunchSettings.json
7. Copy & Paste the console launch settings (commandName: Project) with a different Name
8. Change ASPNETCORE_ENVIRONMENT to "Production"
9. Launch the app with new Launch Setting "Production" again and go to "Register" or "Login" tab
10. See the page is rendered without specific site layout from the css files.
![grafik](https://user-images.githubusercontent.com/39725141/78028407-48483300-735f-11ea-8f58-28b91d0a8bde.png)

Unresolvable:
- https://localhost:5001/Identity/js/site.js
- https://localhost:5001/Identity/css/site.css

I can not figure out why this happens and do not know a workarround. Would be nice if anyone can give me a hint how to circumvent this. Or am I missing something very basic here? (I am quite new to Web Development with Blazor).

**Annotation**:
Some of the ressources can be resolved properly (e.g. Bootstrap), because in the default implementation some script references have environment switches to embed different versions in production. (e.g. _ValidationScriptsPartial.cshtml references ajax.aspnetcdn.com/... instead of ~/Identity/... in Production, which again is not good because the application might not work in a non-internet environment).

**Scaffolding: makes it even worse!**
The effect gets even worse, when using Scaffolded items.
1. Add a new scaffolding for Login and Register page without specifiying an existing layout but with the existing db context
2. Start app again with default Development launch setting and see Login/Register is fine
3. Restart app with Production launch setting, go to Login/Register and see a **complete mess**, since this time also bootstrap and jquery references are not resolvable (they are referenced in _Layout.cshtml also with ~/Identity/ in the path)
![grafik](https://user-images.githubusercontent.com/39725141/78029912-99f1bd00-7361-11ea-9f9b-b3e00e0579b6.png)

**This is how same code looks in Development mode**
![grafik](https://user-images.githubusercontent.com/39725141/78030211-02d93500-7362-11ea-816f-344478091886.png)

** Version note **
I tested this with different 3.x versions of .NET Core and always get the same results.
It still happens on the LTS version 3.1.3. I could not test it with .NET 5 version yet.

### Further technical details
Versions used (latest):
- Visual Studio Enterprise 2019 16.5.1
- netcoreapp3.1
- Microsoft.AspNetCore.Identity.UI 3.1.3

.NET Core SDK (gemäß "global.json"):
 Version:   5.0.100-preview.1.20155.7
 Commit:    1c44853056

Laufzeitumgebung:
 OS Name:     Windows
 OS Version:  10.0.18363
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\5.0.100-preview.1.20155.7\

Host (useful for support):
  Version: 5.0.0-preview.1.20120.5
  Commit:  3c523a6a7a

.NET Core SDKs installed:
  3.1.101 [C:\Program Files\dotnet\sdk]
  3.1.200 [C:\Program Files\dotnet\sdk]
  3.1.201 [C:\Program Files\dotnet\sdk]
  3.1.300-preview-015048 [C:\Program Files\dotnet\sdk]
  5.0.100-preview.1.20155.7 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.16 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.16 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.0-preview.1.20124.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.16 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.0-preview.1.20120.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.2 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.3 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.0-preview.1.20127.5 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

**Comment by tiwj** [comment]
@danroth27  any Idea on this? I saw you focused on similar cases related to Blazor and identity

**Comment by javiercn** [comment]
@tiwj thanks for contacting us.

Static web assets is only enabled in development scenarios by default. You can call host.UseStaticWebassets() on your webhost builder unconditionally to enable them. That said, my recommendation is that if you want to run in "Production" you do so against the published output of the app.

**Comment by tiwj** [comment]
> 
> 
> @tiwj thanks for contacting us.
> 
> Static web assets is only enabled in development scenarios by default. You can call host.UseStaticWebassets() on your webhost builder unconditionally to enable them. That said, my recommendation is that if you want to run in "Production" you do so against the published output of the app.

Thank you for the quick reply and shame on my confusion. Both mentioned methods solved the issue. I found out that the actual problem was that **wwwroot/Identity** was not included in the publish output before I updated Visual Studio to the latest version today. 

This led to the issue in my Production environment which then led me to test arround with it inside VS. This seems to be fixed in latest VS Version, since wwwroot/Identity is now included during Publish process (was definitley not the case with Identity UI 3.1.2 and VS 16.4.x. The wwwroot/Identity folder does not exist on my production machine with the output from the 16.4.x version).
